stan/math/rev/functor/coupled_ode_system.hpp

```diff
@@ -119,7 +119,10 @@ struct coupled_ode_system<F, double, var> {
     try {
       start_nested();
 
-      const vector<var> y_vars(z.begin(), z.begin() + N_);
+      vector<var> y_vars;
+      y_vars.reserve(N_);
+      for (std::size_t i = 0; i < N_; ++i)
```

Comment:
I'd strongly prefer this to be done with range iterators, as in:

```
for (double zi : z)
  y_vars.emplace_back(new vari(zi, false));
```

Reply:
No, that does not work, since we do not iterate over the entire z vector. Only the first N elements of it.